openvdb/openvdb/tree/ValueAccessor.h

```diff
@@ -801,6 +826,324 @@ class CacheItem
         }
     }
 
+private:
+    CacheItem(const CacheItem&);
+    CacheItem& operator=(const CacheItem&);
+
+    bool isHashed(const Coord& xyz) const
+    {
+        return (xyz[0] & ~Coord::ValueType(NodeType::DIM-1)) == mHash[0]
+            && (xyz[1] & ~Coord::ValueType(NodeType::DIM-1)) == mHash[1]
+            && (xyz[2] & ~Coord::ValueType(NodeType::DIM-1)) == mHash[2];
+    }
+
+    TreeCacheT* mParent;
+    Coord mHash;
+    const NodeType* mNode;
+    using RestT = typename NodeVecT::PopFront;
+    CacheItem<TreeCacheT, RestT, /*AtRoot=*/RestT::Size == 1, false> mNext;
+};// end of CacheItem
+
+
+#if OPENVDB_ABI_VERSION_NUMBER >= 10
+
+template<typename TreeCacheT, typename NodeVecT>
+class CacheItem<TreeCacheT, NodeVecT, /*AtRoot=*/false, true>
```

Comment:
We unfortunately need an entire new specialization of `CacheItem` for LeafNodes for the general accessor case, otherwise we end up adding an extra pointer to every `CacheItem` unnecessarily.